### PR TITLE
Drop LAPIv2 from released artifacts

### PR DESCRIPTION
**Pull request by nmarton-da**

### CODE REVIEW

File: canton/BUILD.bazel [hunk 1]
```diff
@@ -1274,6 +1274,30 @@ proto_jars(
     srcs = glob(["community/ledger-api/src/main/protobuf/com/daml/ledger/api/v*/**/*.proto"]),
     grpc = True,
     java_conversions = True,
+    maven_artifact_prefix = "ledger-api-internal",
```
Comment (nmarton-da) [inline canton/BUILD.bazel:1277]: this is not getting released anymore, it is for internal usage with lapiv2 (lot of dependencies in ledger api implementation)

File: canton/BUILD.bazel [hunk 2]
```diff
@@ -1274,6 +1274,30 @@ proto_jars(
...
+        "@com_google_protobuf//:timestamp_proto",
+        "@com_google_protobuf//:wrappers_proto",
+        "@go_googleapis//google/rpc:errdetails_proto",
+        "@go_googleapis//google/rpc:status_proto",
+    ],
+)
+
+proto_jars(
+    name = "ledger_api_proto_external",
+    srcs = glob(["community/ledger-api/src/main/protobuf/com/daml/ledger/api/v1/**/*.proto"]),
```
Comment (nmarton-da) [inline canton/BUILD.bazel:1298]: this is the one which is being released now, filtered to v1

File: canton/BUILD.bazel (hunk 1 above)
Comment (mziolekda) [reply canton/BUILD.bazel:1277]: If there are dependencies on the v2 in 2.x, I would expect somewhere the changes that latch onto this new published artifact and use it :thinking:

Comment (nmarton-da) [reply canton/BUILD.bazel:1277]: I can't think of many usage: canton does not support it anymore (no x nodes). Canton internals use it, but outside should not.
Please note: this is not published, just a necessary compile dependency for the code drop.

Comment (mziolekda) [reply canton/BUILD.bazel:1277]: Oh, the `maven_artifact_prefix` tricked me

**Review by mziolekda — APPROVED** [review]

File: canton/BUILD.bazel (hunk 1 above)
Comment (nmarton-da) [reply canton/BUILD.bazel:1277]: yes unfortunately that is needed by our proto support (fails in pom generation), tried to do it without at first, then I realized we anyway declare all the publishable artifacts in artifacts.yaml

**Review by filmackay — APPROVED** [review]

**Review by remyhaemmerle-da — APPROVED** [review]

**Pull request merged**